### PR TITLE
apps/testcase/kernel: changes in ITC of pthread in itc_pthread.c

### PR DESCRIPTION
**Pull request by manoj11588612**
testcase :
itc_pthread_key_create_set_get_specific_p_multitime -> call of create key and destroy key in subroutine.

Signed-off-by: manoj <manoj.g2@samsung.com>

### CODE REVIEW

**Comment by manoj11588612** [comment]
@sangwon03 @sunghan-chang @samegautam 
Please review

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c [hunk 1]
```diff
@@ -114,7 +120,7 @@ static void *thread_set_get_specific(void *param)
 			return NULL;
 		}
 	}
-	sleep(WAIT_TIME_1);
+	pthread_key_delete(exec_index);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_pthread.c]: 1. exec_index -> g_key
2. Don't you need to check return value of delete?

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c [hunk 2]
```diff
@@ -101,7 +101,13 @@ static void *thread_set_get_specific(void *param)
 	int exec_index;
 	g_flag = false;
 
+	ret_chk = pthread_key_create(&g_key, NULL);
+		if (ret_chk != OK) {
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_pthread.c]: Please check indentation.

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c [hunk 3]
```diff
@@ -329,22 +335,16 @@ static void itc_pthread_mutex_init_destroy_p_multitime(void)
...
+static void itc_pthread_key_create_set_get_specific_p_multitime(void)
 {
 	int ret_chk;
 	g_flag = false;
-	g_key = 0;
-
-	ret_chk = pthread_key_create(&g_key, NULL);
-	TC_ASSERT_EQ("pthread_key_create", ret_chk, OK);
 
 	ret_chk = pthread_create(&g_pthread, NULL, thread_set_get_specific, NULL);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_pthread.c]: thread_set_get_specific should be changed to include create and delete.
I think itc_pthread_key_test is enough.

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c [hunk 4]
```diff
@@ -329,22 +335,16 @@ static void itc_pthread_mutex_init_destroy_p_multitime(void)
 * Preconditions         :pthread_key_create
 * Postconditions        :none
 */
-static void itc_pthread_key_create_set_getspecific_p_multitime(void)
+static void itc_pthread_key_create_set_get_specific_p_multitime(void)
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_pthread.c]: Please add "delete" also.

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c [hunk 5]
```diff
@@ -101,7 +101,13 @@ static void *thread_set_get_specific(void *param)
 	int exec_index;
 	g_flag = false;
 
+	ret_chk = pthread_key_create(&g_key, NULL);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_pthread.c]: Don't you need to put this into for loop?

**Comment by seinfra** [comment]
Target : [58955af3a2885db6c301ebada92528b9033a42d4] -

**Comment by seinfra** [comment]
Target : [58955af3a2885db6c301ebada92528b9033a42d4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [58955af3a2885db6c301ebada92528b9033a42d4] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c (hunk 1 above)
Comment (manoj11588612) [reply apps/examples/testcase/le_tc/kernel/itc_pthread.c]: added return check for pthread_key_delete  API

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c (hunk 2 above)
Comment (manoj11588612) [reply apps/examples/testcase/le_tc/kernel/itc_pthread.c]: done

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c (hunk 3 above)
Comment (manoj11588612) [reply apps/examples/testcase/le_tc/kernel/itc_pthread.c]: changed subroutine name as you mentioned

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c (hunk 4 above)
Comment (manoj11588612) [reply apps/examples/testcase/le_tc/kernel/itc_pthread.c]: done

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c (hunk 5 above)
Comment (manoj11588612) [reply apps/examples/testcase/le_tc/kernel/itc_pthread.c]: added both create and delete into loop

**Comment by seinfra** [comment]
Target : [ef89fc6749b2c85e97fbf107d8d76232fe5a51c7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ef89fc6749b2c85e97fbf107d8d76232fe5a51c7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ef89fc6749b2c85e97fbf107d8d76232fe5a51c7] -

File: apps/examples/testcase/le_tc/kernel/itc_pthread.c [hunk 6]
```diff
@@ -113,8 +119,14 @@ static void *thread_set_get_specific(void *param)
 			printf("pthread_getspecific FAIL\n");
 			return NULL;
 		}
+
+		ret_chk = pthread_key_delete(exec_index);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/itc_pthread.c]: You didn't change here. ```exec_index -> g_key```

**Comment by seinfra** [comment]
Target : [f7a5613829d6be70937fb3d9abaa28134faad503] -

**Comment by seinfra** [comment]
Target : [f7a5613829d6be70937fb3d9abaa28134faad503] - Code Rule Check OK.

**Pull request merged**